### PR TITLE
feat: add sveltekit auth example (#801)Co-authored-by: Eelco Wiersma <contact@pagebakers.nl> Co-authored-by: Dani Akash <DaniAkash@users.noreply.github.com>

### PR DESCRIPTION
**Pull request by leoalves**
## Motivation and Context

Add Wundergraph cookie based authentication example with SvelteKit

#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: examples/sveltekit-oidc-cookies-auth/package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+	"name": "sveltekit-oidc-cookies-auth",
```
Comment (Pagebakers) [inline examples/sveltekit-oidc-cookies-auth/package.json:2]: Please prefix the name with `wundergraph-`

File: examples/sveltekit-oidc-cookies-auth/src/lib/components/Navbar.svelte [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import { page } from '$app/stores';
+	import { createWGClient } from '$lib/svelte-query';
+
+	const { getAuth, createQuery } = createWGClient();
```
Comment (Pagebakers) [inline examples/sveltekit-oidc-cookies-auth/src/lib/components/Navbar.svelte:6]: I'd initiate the client in lib/svelte-query already so you can simply import getAuth / createQuery in components and routes.

Is there any specific reason why you decided for this with Svelte?

File: examples/sveltekit-oidc-cookies-auth/src/lib/components/UserFromCookie.svelte [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<script lang="ts">
+	import { createWGClient } from '$lib/svelte-query';
+	const { getUser } = createWGClient();
```
Comment (Pagebakers) [inline examples/sveltekit-oidc-cookies-auth/src/lib/components/UserFromCookie.svelte:3]: Idem, import getUser from $lib/svelte-query

File: examples/sveltekit-oidc-cookies-auth/src/routes/+page.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+export const load: PageLoad = async ({ parent, fetch }) => {
+  const { queryClient } = await parent()
+  
+  const {queryKey} = createWGClient(fetch)
+  const client = createClient({
+    customFetch: fetch
+  })  
+
+  await queryClient.prefetchQuery({
```
Comment (Pagebakers) [inline examples/sveltekit-oidc-cookies-auth/src/routes/+page.ts:13]: Will look into your other PR to add this to the client asap.

**Review by Pagebakers — CHANGES_REQUESTED** [review]
Awesome work and great showcase for WunderGraph! I left a few comments.

File: examples/sveltekit-oidc-cookies-auth/src/lib/components/Navbar.svelte (hunk 2 above)
Comment (leoalves) [reply examples/sveltekit-oidc-cookies-auth/src/lib/components/Navbar.svelte:6]: This was a mistake from my part.  I was trying a few different things with the customFetch function provided by SvelteKit and didn't change it afterwards.  I will change it now.

**Review by Pagebakers — APPROVED** [review]
LGTM!

**Comment by Pagebakers** [comment]
Can you update the example to use the new prefetchQuery method?
https://docs.wundergraph.com/docs/clients-reference/svelte-query#ssr

**Comment by leoalves** [comment]
> Can you update the example to use the new prefetchQuery method? https://docs.wundergraph.com/docs/clients-reference/svelte-query#ssr

I updated the code. But I don't think the changes are updated in the NPM repository.

File: examples/sveltekit-oidc-cookies-auth/package.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		"vite": "^4.2.0",
+		"vitest": "^0.25.3",
+		"wait-on": "^7.0.1"
+	},
+	"type": "module",
+	"dependencies": {
+		"@prisma/client": "^4.12.0",
+		"@tanstack/svelte-query": "^4.27.0",
+		"@wundergraph/sdk": "^0.143.0",
+		"@wundergraph/svelte-query": "^0.1.0"
```
Comment (DaniAkash) [inline examples/sveltekit-oidc-cookies-auth/package.json:56]: Changing this svelte query package version to latest should be enough to merge this PR :D 
```suggestion
		"@wundergraph/svelte-query": "^0.2.2"
```

**Review by DaniAkash — APPROVED** [review]

**Review by DaniAkash — APPROVED** [review]

**Comment by leoalves** [comment]
> CI is failing https://github.com/wundergraph/wundergraph/actions/runs/4744045831/jobs/8424403243?pr=801 @leoalves could you take a look?

I believe the issue was a mismatch in the `@tanstack/svelte-query` version installed in this package and the one from `@wundergraph/svelte-query` it should be good now.

**Comment by Pagebakers** [comment]
Getting the following error in the tests:

```
Warning: G] Cannot find base config file "./.svelte-kit/tsconfig.json" [tsconfig.json]

    tsconfig.json:2:12:
      2 │   "extends": "./.svelte-kit/tsconfig.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by leoalves** [comment]
> Getting the following error in the tests:
> 
> ```
> Warning: G] Cannot find base config file "./.svelte-kit/tsconfig.json" [tsconfig.json]
> 
>     tsconfig.json:2:12:
>       2 │   "extends": "./.svelte-kit/tsconfig.json",
>         ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

Tthat is just a warning.  That tsconfig.json is generated afterwards when running the `build`, `dev` or `sync` commands. 

The error is coming from the QueryClient type (same error as before).  It's comparing types from 2 different versions of the `@tanstack/svelte-query` package. One from this example and the other one installed in the `@wundergraph/svelte-query`. 
Both types are exactly the same and when running the check command locally `npm run check` I don't get any errors.
I can try to to copy the `@tanstack/svelte-query` version from the `@wundergraph/svelte-query` package into this example `package.json`.  But I am not sure if that will solve the problem

**Pull request merged**